860 sql files generation

quantumfate opened the pull request:
Adresses #860 .

Implements

* SQL Script generation based on entity
* SQL utility class

@devonfw/cobigen

Commit:
#860 template structure and sql util class

Commit:
Refactored Util classes, implemented sql util functions

Commit:
removed .classpath from tracking

Commit:
Helper function mapJavaToSqlType(String canonicalTypeName) implemented

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/devonfw/cobigen?pullRequest=1570) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/devonfw/cobigen?pullRequest=1570) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: Zylesto<br/>:white_check_mark: jan-vcapgemini<br/>:white_check_mark: Lur1an<br/>:white_check_mark: mdukhan<br/>:x: CORP\leholm<br/><hr/>**CORP\leholm** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/devonfw/cobigen?pullRequest=1570) it.</sub>

Commit:
Added a function that allows to retrieve annotations

Commit:
Added a new dependency for SQL specific annotations, almost finished getSqlType

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/SQLUtil.java:
```diff
@@ -0,0 +1,322 @@
+   * @return type of the field in the string
+   * @throws ClassNotFoundException
+   */
+  @SuppressWarnings("javadoc")
```
Do we need that?

Commit:
Implemented Varchar, Not Null and Auto Increment annotation

Commit:
Functions for non collections

Commit:
Implemented Foreign Key Functions and template

Commit:
Improved javadoc, implemented mappedby aspect into foreignkey method, removed unnecessary List wrapper around HashMaps

Commit:
Implemented getSImpleSQLtype and Primary key methods

Commit:
Added a default parameter to getForeignKeyData and fixed bug in getForeignKeyStatement

Commit:
getForeignKeyStatement and template - added Test class for SQL Util

Commit:
Testclass for SQL

Commit:
Added test classes

Commit:
Added test cases and refined getsimplesql statement

Commit:
More tests

Commit:
Implemented utility tests and refined methods

Commit:
Primary key method refined and tests implemented

Commit:
Primary key tests

Commit:
Foreign Key methods

Commit:
implemented isFieldEntity

Commit:
Merge branch 'master' of github.com:devonfw/cobigen into 860_SQL_files_generation

Commit:
comments nad pulled upstream master

Lur1an committed:
- Added JavaPlugin dependency to template project for testing purposes
- Implemented abstract freemarker test for java class inputs
- Implemented barebone SQL generation test
- Rework of SQLUtil and Entity.sql template

Lur1an committed:
Finished blueprint for template

Lur1an committed:
Implemented most functionality for template generation. Tests are lacking

Lur1an committed:
Simplified usage of AbstractJavaTemplateTest.
Reduced functionality to process(JavaClass) method

Lur1an committed:
Merge branch 'master' into 860_SQL_files_generation

Lur1an committed:
Fixes to Nullable and Unique constraint parsing for @Column and @JoinColumn annotations

Lur1an committed:
Fixed logic error pertaining @OneToMany and @ManyToOne annotations

Lur1an committed:
Implemented parsing of @JoinTable annotation into a separate inline SQL Table for the ManyToMany relationships.

Zylesto committed:
#860 Added first tests for sql generation

Zylesto committed:
#860 added more tests, fixed bugs

Lur1an committed:
- Removed some unneccessary Blob/Clob logic
- Fixed test assertions
- Fixed SQL Type mapping regex and matching order
- Implemented test for @JoinTable and @ManyToMany

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/AbstractJavaTemplateTest.java:
```diff
@@ -0,0 +1,117 @@
+   * Model used by freemarker
+    */
+  public Map<String, Object> model;
+  public FreeMarkerTemplateEngine engine;
```
Javadoc missing.

jan-vcapgemini commented on cobigen-templates/sql-openapi-app/context.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<contextConfiguration xmlns="http://capgemini.com/devonfw/cobigen/ContextConfiguration"
+                      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                      version="3.0">
+    <trigger id="sql_openapi_app" type="openapi">
+
+    </trigger>
+    <tags>
+      <tag name="OpenAPI"></tag>
+      <tag name="SQL"></tag>
+      <tag name="App"></tag>
+    </tags>
+</contextConfiguration>
```
File and folder containing it can be removed?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/.gitignore:
```diff
@@ -0,0 +1 @@
+.classpath
```
Please remove this change.

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/SQLTemplateGenerationTest.java:
```diff
@@ -0,0 +1,75 @@
+  @Override
+  public String getTemplatePath() {
+
+    return "src/main/templates/sql_java_app/templates/V0000__Create_${variables.entityName}Entity.sql.ftl";
```
Shouldn't this be moved to a constant instead of using a getter?

Lur1an replied:
The getter of the abstract class needs to be overridden for it to function.

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/primarykeys/Primarytwo.java:
```diff
@@ -0,0 +1,18 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.primarykeys;
+
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * TODO leholm This type ...
+ *
+ */
+public class Primarytwo {
+  @Id
+  @Column(name = "TEST_ID", length = 50, nullable = false)
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private Long id;
+
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/primarykeys/Primarythree.java:
```diff
@@ -0,0 +1,17 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.primarykeys;
+
+import javax.persistence.Id;
+
+/**
+ * TODO leholm This type ...
+ *
+ */
+public class Primarythree {
+  private Long testId;
+
+  @Id
+  public Long getTestId() {
+
+    return this.testId;
+  }
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/primarykeys/Primaryone.java:
```diff
@@ -0,0 +1,12 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.primarykeys;
+
+import javax.persistence.Id;
+
+/**
+ * TODO leholm This type ...
+ *
+ */
+public class Primaryone {
+  @Id
+  Long id;
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/primarykeys/Primaryfour.java:
```diff
@@ -0,0 +1,19 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.primarykeys;
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+
+/**
+ * TODO leholm This type ...
+ *
+ */
+public class Primaryfour {
+  private Long testId;
+
+  @Id
+  @Column(name = "TEST_ID", length = 50, nullable = false)
+  public Long getTestId() {
+
+    return this.testId;
+  }
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/primarykeys/Primaryfive.java:
```diff
@@ -0,0 +1,14 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.primarykeys;
+
+/**
+ * TODO leholm This type ...
+ *
+ */
+public class Primaryfive {
+  private Long id;
+
+  public Long getId() {
+
+    return this.id;
+  }
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/TestCat.java:
```diff
@@ -0,0 +1,36 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest;
+
+/**
+ * This is a child animal class for testing purposes.
+ *
+ */
+public class TestCat extends TestAnimal {
+
+  private Integer legs;
+
+  /**
+   * The constructor.
+   */
+  public TestCat(String name, Integer legs) {
+
+    super(name);
+    this.legs = legs;
+  }
+
+  /**
+   * @return legs
+   */
+  public Integer getLegs() {
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/TestSqlTypeAnnotations.java:
```diff
@@ -0,0 +1,116 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest;
+
+import javax.persistence.Column;
+import javax.persistence.Id;
+
+import com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.entities.TestAnotherSimpleEntity;
+import com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.entities.TestNotSoSimpleEntity;
+import com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.entities.TestSimpleEntity;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+/**
+ * This class is a test class for {@link SQLAnnotationTest}
+ *
+ */
+public class TestSqlTypeAnnotations {
+
+  // Specific Statement variations
+  @Id
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/entities/TestAnotherSimpleEntity.java:
```diff
@@ -0,0 +1,49 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.entities;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * This is a simple Entity Class for testing.
+ *
+ */
+
+@Entity
+@Table(schema = "RECORDS")
+public class TestAnotherSimpleEntity {
+
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/entities/TestNotSoSimpleEntity.java:
```diff
@@ -0,0 +1,47 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.entities;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * This is a simple Entity Class for testing.
+ *
+ */
+
+@Entity
+public class TestNotSoSimpleEntity {
+
+  @Id
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/enums/TestSimpleEnum.java:
```diff
@@ -0,0 +1,11 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest.enums;
+
+/**
+ * This is an enum for sql testing purposes
+ *
+ */
+@SuppressWarnings("javadoc")
+public enum TestSimpleEnum {
+
+  ONE, TWO, THREE
+}
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/resources/sqltest/TestAnimal.java:
```diff
@@ -0,0 +1,46 @@
+package com.devonfw.cobigen.templates.devon4j.test.utils.resources.sqltest;
+
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+/**
+ * This is a parent class for all animals for testing purposes.
+ *
+ */
+public class TestAnimal {
```
Is this file still needed / used?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/AbstractJavaTemplateTest.java:
```diff
@@ -0,0 +1,117 @@
+
+  /**
+   * Adds object to the model
+   * @param key
+   * @param instance
```
Javadoc description for params missing.

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/AbstractJavaTemplateTest.java:
```diff
@@ -0,0 +1,117 @@
+
+  /**
+   * Creates a template engine for the given template path
+   * 
+   * @param templateFolderPath Relative path to template folder from project source root
+   */
+  public FreeMarkerTemplateEngine createEngine(String templateFolderPath) {
```
Javadoc for return missing.

Lur1an replied:
Is return type and method name not self-explanatory?

jan-vcapgemini commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/AbstractJavaTemplateTest.java:
```diff
@@ -0,0 +1,117 @@
+  /**
+   * Creates an anonymous TextTemplate object that works on the given template files
+   * 
+   * @param relativePath Relative Path to the template from the /templates folder, most likely coincides with template
+   *        filename
+   * @param relativeAbsolutePath Relative path from source root to the template
+   */
+  public TextTemplate createTemplate(String relativePath, Path relativeAbsolutePath) {
```
Javadoc for return missing.

Lur1an committed:
Adressing change-request:
- removal of dead classes/files
- addition of javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/JavaUtil.java:
```diff
@@ -215,31 +207,6 @@ public String castJavaPrimitives(Class<?> pojoClass, String fieldName)
     }
   }
 
-  /**
```
The description ins missing. What does this mehtod?

Zylesto commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/JavaUtil.java:
```diff
@@ -215,31 +207,6 @@ public String castJavaPrimitives(Class<?> pojoClass, String fieldName)
-   * @param pojoClass {@link Class} the class object of the pojo
-   * @param fieldName {@link String} the name of the field
-   * @return true if the field is an instance of java.utils.Collections
-   * @throws NoSuchFieldException indicating something awefully wrong in the used model
```
awfully

Zylesto commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/JavaUtil.java:
```diff
@@ -424,22 +391,6 @@ private Method findMethod(Class<?> pojoClass, String methodName) {
     return null;
```
Is this correctly implemented?

Lur1an replied:
Not code pertaining to this PR.

Zylesto commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/JavaUtil.java:
```diff
@@ -215,31 +207,6 @@ public String castJavaPrimitives(Class<?> pojoClass, String fieldName)
-  }
-
   /**
    * Returns the Ext Type to a given java type
```
params and return are missing at the javadocs

Lur1an replied:
These methods are not adressed in this pull-request

Lur1an replied:
Not code pertaining to this PR.

Zylesto commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/SQLUtil.java:
```diff
@@ -0,0 +1,298 @@
+
+  /**
+   * Extracts the name of the field from the Map whilst checking for name-override in @Column annotation
+   */
```
param and return are missing in the jvadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/SQLTemplateGenerationTest.java:
```diff
@@ -0,0 +1,75 @@
+    assertThat(output).contains("test_id BIGINT, FOREIGN KEY (test_id) REFERENCES SQLTEST(MY_ID_FIELD)");
+  }
+
+  @Test
```
Javadoc is missing

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/ReferenceEntity.java:
```diff
@@ -0,0 +1,19 @@
+
+import javax.persistence.*;
+
+@Entity
```
Maybe include a javadoc

Lur1an replied:
I don't think putting javadoc in the test entities is necessary

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestDataTypesEntity.java:
```diff
@@ -0,0 +1,388 @@
+public class SQLTestDataTypesEntity {
+
+  @Column()
+  private int integer1;
```
I dont know if it is necessary here, but maybe add javadocs to the fields

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+public class SQLTestEntity {
+  @Id
+  @Column(name = "MY_ID_FIELD")
+  private Long id;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+  private Long id;
+
+  @Column(name = "VALUENAME")
+  private Integer integerValue;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+
+  @Enumerated(EnumType.STRING)
+  @Column(length = 420, name = "ENUM_TEST_FIELD_NAME_OVERRIDE")
+  private EnumForTest enumForTest;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+  @Column(length = 420, name = "ENUM_TEST_FIELD_NAME_OVERRIDE")
+  private EnumForTest enumForTest;
+
+  public Long getId() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+    return this.id;
+  }
+
+  public void setId(Long id) {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+    this.id = id;
+  }
+
+  public Integer getIntegerValue() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+    return this.integerValue;
+  }
+
+  public void setIntegerValue(Integer value) {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+    this.integerValue = value;
+  }
+
+  public EnumForTest getEnumForTest() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestEntity.java:
```diff
@@ -0,0 +1,58 @@
+    return this.enumForTest;
+  }
+
+  public void setEnumForTest(EnumForTest enumForTest) {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestForeignKeysEntity.java:
```diff
@@ -0,0 +1,61 @@
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  @Column(name = "id")
+  private Long id;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestForeignKeysEntity.java:
```diff
@@ -0,0 +1,61 @@
+
+  @OneToOne()
+  @JoinColumn(referencedColumnName = "MY_ID_FIELD", name = "test_id")
+  private SQLTestEntity sqlTestEntity;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.Set;
+
+@Entity
+public class SQLTestJoinTableEntity {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+public class SQLTestJoinTableEntity {
+    @Id
+    @Column(name = "id", nullable = false)
+    private Long id;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+            joinColumns = @JoinColumn(name = "REF_ENTITY_ID", table = "REFERENCE", unique = true, referencedColumnName = "OVERRIDE_ID"),
+            inverseJoinColumns = @JoinColumn
+    )
+    private Set<ReferenceEntity> referenceEntities;
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+    )
+    private Set<ReferenceEntity> referenceEntities;
+
+    public Long getId() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+        return id;
+    }
+
+    public void setId(Long id) {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+        this.id = id;
+    }
+
+    public Set<ReferenceEntity> getReferenceEntities() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestJoinTableEntity.java:
```diff
@@ -0,0 +1,35 @@
+        return referenceEntities;
+    }
+
+    public void setReferenceEntities(Set<ReferenceEntity> referenceEntities) {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class SQLUtilTest {
+
+  @Test
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  @Test
+  public void testByteArray() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  @Test
+  public void testTimestamp() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  @Test
+  public void testTimeZone() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  }
+
+  @Test
+  public void testCalendar() {
```
Missing javadoc

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  @Test
+  public void testByteArray() {
+
+    // Test fails
```
Why are there missing tests?

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  @Test
+  public void testTimestamp() {
+
+    // Test fails
```
Why are there failing tests?

Zylesto commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/SQLUtilTest.java:
```diff
@@ -0,0 +1,50 @@
+  @Test
+  public void testTimeZone() {
+
+    // Test fails
```
Why are there failing tests?

Lur1an replied:
The tests don't fail, I'm removing the comments.

Zylesto reviewed (changes requested):
Please add more JavaDocs and remove failing tests

Lur1an committed:
Removed unneccessary comments and added javadoc to test

Lur1an committed:
added jdoc to getValue() method in SQLUtil

NelsonNew commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/CommonUtil.java:
```diff
@@ -0,0 +1,70 @@
+   * @param pojoClass {@link Class} the class object of the pojo
+   * @param fieldName {@link String} the name of the field
+   * @return true if the field is an instance of java.utils.Collections
+   * @throws NoSuchFieldException indicating something awefully wrong in the used model
```
Typo "awefully"

NelsonNew commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestDataTypesEntity.java:
```diff
@@ -0,0 +1,388 @@
+  private byte[] blob;
+
+   @Column()
+   private Class<?> varchar2;
```
Auto-Format

NelsonNew commented on cobigen-templates/templates-devon4j/src/test/java/com/devonfw/cobigen/templates/devon4j/test/templates/testclasses/SQLTestDataTypesEntity.java:
```diff
@@ -0,0 +1,388 @@
+   public void setVarchar2(Class<?> varchar2) {
+
+   this.varchar2 = varchar2;
+   }
```
Auto-Format

Lur1an committed:
Adressed change requests

Lur1an committed:
Deleted CommonUtil.java and restored JavaUtil to previous state.

KHucklenbroich commented on cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/SQLUtil.java:
```diff
@@ -0,0 +1,300 @@
+ * Provides operations to identify and process SQL specific information
+ *
+ */
+public class SQLUtil {
```
I don't like the name of the class. Util classes are static

jan-vcapgemini committed:
Merge branch 'master' into 860_SQL_files_generation

jan-vcapgemini committed:
Merge branch 'master' into 860_SQL_files_generation

Commit:
Merge branch 'master' into 860_SQL_files_generation

# Conflicts:
#	cobigen-eclipse/cobigen-eclipse-test/.classpath

mdukhan committed:
Merge branch 'master' into 860_SQL_files_generation

mdukhan committed:
Merge branch 'master' into 860_SQL_files_generation

Commit:
no message

quantumfate reviewed (commented):
Changes are waiting for approval.
